java/org/apache/tomcat/util/net/TLSClientHelloExtractor.java

```diff
@@ -193,6 +214,15 @@ public String getSNIValue() {
     }
 
 
+    public List<String> getClientRequestedCipherNames() {
+        if (result == ExtractorResult.COMPLETE || result == ExtractorResult.NOT_PRESENT) {
+            return clientRequestedCipherNames;
+        } else {
+            throw new IllegalStateException();
```

Comment:
Intentionally without a message ?

Reply:
-1 to having no message

Reply:
I'll add a message. Force push to follow shortly.